sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.controller.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable, NotFoundException } from "@nestjs/common";
+import { ApplicationOfferingChangeRequestService } from "../../services";
+import { ApplicationOfferingChangesAPIOutDTO } from "./models/application-offering-change-request.dto";
+import { getUserFullName } from "../../utilities";
+
+@Injectable()
+export class ApplicationOfferingChangeRequestControllerService {
+  constructor(
+    private readonly applicationOfferingChangeRequestService: ApplicationOfferingChangeRequestService,
+  ) {}
+
+  /**
+   * Get the Application Offering Change Request by its id.
+   * @param id the Application Offering Change Request id.
+   * @param options method options:
+   * - `studentId`: student id for student authorization.
+   * - `locationId`: location id for institution authorization.
+   * @returns application offering change request.
+   */
+  async getById(
+    id: number,
+    options?: {
+      studentId?: number;
+      locationId?: number;
+    },
+  ): Promise<ApplicationOfferingChangesAPIOutDTO> {
+    const request = await this.applicationOfferingChangeRequestService.getById(
+      id,
+      { studentId: options?.studentId },
+    );
+    if (!request) {
+      throw new NotFoundException(
+        "Not able to find an Application Offering Change Request.",
+      );
+    }
+    return {
```

Comment:
I am seeing a lot more fields are returning than required, Why do need non-required fields? is it a matter of creating a client-specific DTO
![image](https://github.com/bcgov/SIMS/assets/77353155/2fa201f5-f1d6-47c8-bb86-b17b63604bdb)
let me know, if I am missing anything

Reply:
I believe only the below fields are required for students,
![image](https://github.com/bcgov/SIMS/assets/77353155/71163048-4f71-46eb-bf7d-d3e00f7ebb79)